### PR TITLE
quiche: Remove unnecessary methods from QUICHE Flags

### PR DESCRIPTION
**Pull request by RyanTheOptimist**
quiche: Remove unnecessary methods from QUICHE Flags
help() is not used at all, and name() is only used once in a place where it can be avoided.
This also means that the non-templated Flags class can be completely replaced by
the templated TypedFlag.
Finally, the setReloadableValue() method is no longer needed now that
QUICHE reloadable flag behavior matches runtime guard behavior.

Signed-off-by: Ryan Hamilton <rch@google.com>

Risk Level: Low 
Testing: Existing tests
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

**Comment by RyanTheOptimist** [comment]
@danzh2010 I've update this PR now that the other PR has landed. PTAL.

File: source/common/quic/platform/quiche_flags_impl.h [hunk 1]
```diff
@@ -17,29 +17,10 @@ const std::string EnvoyQuicheReloadableFlagPrefix =
...
-  std::string help_;
-};
-
 // Concrete class for QUICHE protocol and feature flags, templated by flag type.
-template <typename T> class TypedFlag : public Flag {
+template <typename T> class TypedFlag {
 public:
-  TypedFlag(const char* name, T default_value, const char* help)
-      : Flag(name, help), value_(default_value), default_value_(default_value) {}
+  TypedFlag(T default_value) : value_(default_value), default_value_(default_value) {}
```
Comment (danzh2010) [inline source/common/quic/platform/quiche_flags_impl.h:23]: missing explicit modifier?

Comment (RyanTheOptimist) [reply source/common/quic/platform/quiche_flags_impl.h]: Oh good point. Done!

**Review by danzh2010 — APPROVED** [review]
LGTM

**Review by KBaichoo — APPROVED** [review]
lgtm, thanks

**Pull request merged**